src/types/media-session.ts

```diff
@@ -0,0 +1,103 @@
+import { MediaSession } from 'stanza/jingle';
+import { JingleAction, JINGLE_INFO_ACTIVE } from 'stanza/Constants';
+import StatsGatherer, { StatsEvent } from 'webrtc-stats-gatherer';
+import StrictEventEmitter from 'strict-event-emitter-types';
+import { EventEmitter } from 'events';
+import { applyMixins } from '../utils';
+import { JingleReason, JingleInfo } from 'stanza/protocol';
+
+export type SessionType = 'softphone' | 'screenShare' | 'screenRecording' | 'collaborateVideo' | 'unknown';
+
+export class GenesysCloudMediaSession extends MediaSession {
+  private statsGatherer?: StatsGatherer;
+
+  constructor (options: any, public sessionType: SessionType, private allowIPv6: boolean) {
+    super(options);
+
+    if (!options.optOutOfWebrtcStatsTelemetry) {
+      this.setupStatsGatherer();
+    }
+  }
+
+  setupStatsGatherer () {
+    this.statsGatherer = new StatsGatherer(this.pc);
+    this.statsGatherer.on('stats', this.emit.bind(this, 'stats'));
+  }
+
+  onIceStateChange () {
+    const state = this.pc.iceConnectionState;
+
+    if (state === 'connected') {
+      this._log('info', 'sending session-info: active');
```

Comment:
If this log is getting sent to Sumo, is there any identifying info we can send with it like a sessionId or convoId?

Reply:
we'll need to go through all the logs in streaming client. I'd like to also add the genesys-cloud-client-logger to streaming client as soon as we get this stanza stuff straightened out.